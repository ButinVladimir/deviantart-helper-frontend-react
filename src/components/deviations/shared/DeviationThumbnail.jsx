import React from 'react';
import PropTypes from 'prop-types';
import Media from 'react-bulma-components/lib/components/media';
import Tag from 'react-bulma-components/lib/components/tag';

export default function DeviationThumbnail({
  title,
  thumbnail,
  url,
  publishedTime,
  nsfw,
}) {
  const publishedTimeDate = new Date(publishedTime);

  return (
    <Media.Item position="left" className="image-container">
      <div>
        <img
          alt={title}
          src={thumbnail.src}
        />
      </div>
      <div>
        <small className="has-text-grey">
          {publishedTimeDate.toLocaleString()}
        </small>
      </div>
      <div>
        <a href={url}>Visit DA page</a>
      </div>
      {nsfw && <Tag color="danger">NSFW</Tag>}
    </Media.Item>
  );
}

DeviationThumbnail.propTypes = {
  title: PropTypes.string.isRequired,
  thumbnail: PropTypes.shape({
    src: PropTypes.string.isRequired,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
  }).isRequired,
  url: PropTypes.string.isRequired,
  publishedTime: PropTypes.number.isRequired,
  nsfw: PropTypes.bool.isRequired,
};
