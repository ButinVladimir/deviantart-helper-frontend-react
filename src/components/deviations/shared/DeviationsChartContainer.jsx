import { connect } from 'react-redux';
import Chart from '../../chart/Chart';
import * as deviationChart from '../../../redux/action-creators/deviations/chart/change-form-field-values';
import { dataSetOptions } from '../../../consts/data-sets';

/**
 * @description
 * Maps Redux state to props.
 *
 * @param {State} state - The Redux state.
 * @returns {Object} Props.
 */
const mapStateToProps = state => ({
  dataSetTitlesMap: new Map(dataSetOptions),
  dataSet: state.deviations.chart.dataSet,
  roundPeriod: state.deviations.chart.roundPeriod,
  showTime: state.deviations.chart.showTime,
  showDifferences: state.deviations.chart.showDifferences,
});

/**
 * @description
 * Maps Redux dispatch to props.
 *
 * @param {Function} dispatch - The Redux dispatch.
 * @returns {Object} Props.
 */
const mapDispatchToProps = dispatch => ({
  dataSetChangeHandler:
    e => dispatch(deviationChart.changeDataSetActionCreator(e.target.value)),
  roundPeriodChangeHandler:
    e => dispatch(deviationChart.changeRoundPeriodActionCreator(e.target.value)),
  showTimeChangeHandler:
    e => dispatch(deviationChart.changeShowTimeActionCreator(e.target.checked)),
  showDifferencesChangeHandler:
    e => dispatch(deviationChart.changeShowDifferencesActionCreator(e.target.checked)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Chart);
