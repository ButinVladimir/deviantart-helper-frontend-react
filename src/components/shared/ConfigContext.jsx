import React, { createContext } from 'react';

const ConfigContext = createContext(null);

/**
 * @description
 * Gets config context provider.
 *
 * @returns {import('react').Provider} Provider.
 */
export const getConfigProvider = () => ConfigContext.Provider;

/**
 * @description
 * HOC to inject config.
 *
 * @param {React} Component - React component.
 * @returns {(props: Object) => React} Functional component.
 */
export default Component => props => (
  <ConfigContext.Consumer>
    {config => <Component config={config} {...props} />}
  </ConfigContext.Consumer>
);
