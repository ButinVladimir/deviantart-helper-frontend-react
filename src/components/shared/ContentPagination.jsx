import React from 'react';
import PropTypes from 'prop-types';
import Section from 'react-bulma-components/lib/components/section';
import Container from 'react-bulma-components/lib/components/container';
import Pagination from 'react-bulma-components/lib/components/pagination';

export default function ContentPagination({
  page,
  pageCount,
  showPagination,
  loadPageHandler,
}) {
  return (
    <Section>
      <Container>
        {showPagination && (
          <Pagination
            // Pagination in Bulma starts from 1 while pagination on backend start from 0.
            current={page + 1}
            total={pageCount}
            delta={3}
            onChange={loadPageHandler}
          />
        )}
      </Container>
    </Section>
  );
}

ContentPagination.propTypes = {
  page: PropTypes.number.isRequired,
  pageCount: PropTypes.number.isRequired,
  showPagination: PropTypes.bool.isRequired,
  loadPageHandler: PropTypes.func.isRequired,
};
