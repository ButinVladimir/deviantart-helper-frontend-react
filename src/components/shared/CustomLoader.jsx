import React from 'react';
import Section from 'react-bulma-components/lib/components/section';
import Container from 'react-bulma-components/lib/components/container';
import Level from 'react-bulma-components/lib/components/level';
import Loader from 'react-bulma-components/lib/components/loader';

export default function CustomLoader() {
  return (
    <Section>
      <Container>
        <Level>
          <Level.Item>
            <Loader className="custom-loader" />
          </Level.Item>
        </Level>
      </Container>
    </Section>
  );
}
