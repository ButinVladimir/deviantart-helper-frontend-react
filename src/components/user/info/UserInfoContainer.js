import { connect } from 'react-redux';
import consumeConfig from '../../shared/ConfigContext';
import UserInfo from './UserInfo';
import userLoadInfo from '../../../redux/action-creators/user/load-info';

/**
 * @description
 * Maps Redux state to props.
 *
 * @param {State} state - The state.
 * @returns {Object} Props.
 */
const mapStateToProps = state => ({
  userId: state.user.userId,
  userName: state.user.userName,
  userType: state.user.userType,
  accessTokenExpires: state.user.accessTokenExpires,
  refreshTokenExpires: state.user.refreshTokenExpires,
  fetchDateThreshold: state.user.fetchDateThreshold,
  requestDateThreshold: state.user.requestDateThreshold,
});

/**
 * @description
 * Maps Redux dispatch to props.
 *
 * @param {Function} dispatch - Redux dispatch.
 * @param {Object} ownProps - Component own props.
 * @returns {Object} Props.
 */
const mapDispatchToProps = (dispatch, ownProps) => ({
  fetchUserData: () => dispatch(userLoadInfo(ownProps.config)),
});

export default consumeConfig(connect(mapStateToProps, mapDispatchToProps)(UserInfo));
