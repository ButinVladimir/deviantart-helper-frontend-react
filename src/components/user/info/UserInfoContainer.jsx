import { connect } from 'react-redux';
import UserInfo from './UserInfo';
import userLoadInfoActionCreator from '../../../redux/action-creators/user/load-info';

/**
 * @description
 * Maps Redux state to props.
 *
 * @param {State} state - The state.
 * @returns {Object} Props.
 */
const mapStateToProps = state => ({
  userId: state.user.userId,
  userName: state.user.userName,
  userType: state.user.userType,
  accessTokenExpires: state.user.accessTokenExpires,
  refreshTokenExpires: state.user.refreshTokenExpires,
});

/**
 * @description
 * Maps Redux dispatch to props.
 *
 * @param {Function} dispatch - Redux dispatch.
 * @returns {Object} Props.
 */
const mapDispatchToProps = dispatch => ({
  fetchUserData: (config) => {
    dispatch(userLoadInfoActionCreator(config));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(UserInfo);
