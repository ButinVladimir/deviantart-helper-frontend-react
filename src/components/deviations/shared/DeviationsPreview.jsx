import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import Media from 'react-bulma-components/lib/components/media';
import Content from 'react-bulma-components/lib/components/content';
import Heading from 'react-bulma-components/lib/components/heading';
import { Field, Control, Checkbox } from 'react-bulma-components/lib/components/form';
import DeviationThumbnail from './DeviationThumbnail';
import StatisticsLevel from '../../shared/StatisticsLevel';
import { DEVIATIONS_DETAILS, ID_PARAM } from '../../../consts/routes';

export default function DeviationsPreview({
  thumbnail,
  title,
  url,
  id,
  publishedTime,
  nsfw,
  views,
  favourites,
  comments,
  downloads,
  selected,
  toggleSelectionHandler,
}) {
  return (
    <Media>
      <DeviationThumbnail
        title={title}
        thumbnail={thumbnail}
        url={url}
        publishedTime={publishedTime}
        nsfw={nsfw}
      />
      <Media.Item>
        <Content>
          <Heading size={3}>
            <NavLink to={`${DEVIATIONS_DETAILS.replace(ID_PARAM, id)}`}>
              {title}
            </NavLink>
          </Heading>
        </Content>
        <StatisticsLevel
          titleValuePairs={[['Views', views], ['Favourites', favourites], ['Comments', comments], ['Downloads', downloads]]}
        />
      </Media.Item>
      <Media.Item position="right">
        <Field>
          <Control>
            <Checkbox checked={selected} onChange={() => toggleSelectionHandler(id)} />
          </Control>
        </Field>
      </Media.Item>
    </Media>
  );
}

DeviationsPreview.propTypes = {
  ...DeviationThumbnail.PropTypes,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  views: PropTypes.number.isRequired,
  favourites: PropTypes.number.isRequired,
  comments: PropTypes.number.isRequired,
  downloads: PropTypes.number.isRequired,
  selected: PropTypes.bool.isRequired,
  toggleSelectionHandler: PropTypes.func.isRequired,
};
