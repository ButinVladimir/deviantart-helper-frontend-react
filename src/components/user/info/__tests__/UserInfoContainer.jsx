import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import UserInfoContainer from '../UserInfoContainer';
import { getConfigProvider } from '../../../shared/ConfigContext';
import config from '../../../../config';
import createDefaultState from '../../../../redux/states/state';
import userLoadInfo from '../../../../redux/action-creators/user/load-info';

jest.mock('../../../../redux/action-creators/user/load-info', () => jest.fn(() => ({ type: 'TEST_ACTION' })));

describe('UserInfoContainer', () => {
  it('can be rendered correctly', () => {
    const state = {
      ...createDefaultState(),
      user: {
        userId: 'User Id',
        userName: 'User Name',
        userType: 'regular',
        accessTokenExpires: 1001,
        refreshTokenExpires: 2001,
        fetchDateThreshold: 3001,
        requestDateThreshold: 4001,
      },
    };

    const mockStore = configureStore([thunk]);
    const store = mockStore(state);

    const ConfigProvider = getConfigProvider();
    const wrapper = mount(
      <ConfigProvider value={config}>
        <Provider store={store}>
          <UserInfoContainer />
        </Provider>
      </ConfigProvider>,
    );

    expect(wrapper).toMatchSnapshot();
    expect(userLoadInfo).toHaveBeenCalledWith(config);
    expect(store.getActions()).toEqual([
      { type: 'TEST_ACTION' },
    ]);
  });
});
