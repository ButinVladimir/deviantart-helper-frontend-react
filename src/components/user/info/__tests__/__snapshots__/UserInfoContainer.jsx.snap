// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfoContainer renders correctly 1`] = `
<section
  className="section"
  style={Object {}}
>
  <div
    className="container"
    style={Object {}}
  >
    <div
      className="content"
      style={Object {}}
    >
      <h1
        className="title is-2"
      >
        Current user info
      </h1>
      <p>
        <span>
          User: 
        </span>
        <strong>
          User Name
        </strong>
      </p>
      <p>
        <span>
          User id: 
        </span>
        <strong>
          User Id
        </strong>
      </p>
      <p>
        <span>
          User type: 
        </span>
        <strong>
          regular
        </strong>
      </p>
      <p>
        <span>
          Access token expires: 
        </span>
        <strong>
          1970-1-1 03:00:01
        </strong>
      </p>
      <p>
        <span>
          Refresh token expires: 
        </span>
        <strong>
          1970-1-1 03:00:02
        </strong>
      </p>
      <p>
        <span>
          Fetch date threshold: 
        </span>
        <strong>
          1970-1-1 03:00:03
        </strong>
      </p>
      <p>
        <span>
          Request date threshold: 
        </span>
        <strong>
          1970-1-1 03:00:04
        </strong>
      </p>
    </div>
  </div>
</section>
`;
